### ISSUE TITLE
evaluation

### ISSUE THREAD

**Issue by rimmmmm** [hobbit-project/DEBS-GC-2018#32]
Hi all,
we can evaluate the system using the training dataset itself,
but I would like to use a different file, for sending test streams
the available files include unknown ports not included in the training dataset 
best,

**Comment by smirnp** [comment]
Hi! 
Please find the extended list of ports here: https://hobbitdata.informatik.uni-leipzig.de/DEBS_GC_2018/ports_extended.csv 

Currently I'm splitting long trips by ports from the extended list. 
The updated training dataset splitted by extended port list will be announced in the next couple of days.

**Comment by nduy** [comment]
Hi Pavel,
Thank you for your hard working. I found that the number of ports is now 40, instead of 25 as it used to be. 
1) Could you please let me know why these ports appeared? Are they middle points of the current trajectories, and now you want to use them to split?

2) Is there any chance that a port does not appear in training data (as departure and/or arrival port) will be in test data set?

3) As number of port increased, there may rise the problem of insufficient training data. Currently, there are about 400+ trajectories for 25 ports. With 40 ports. if the number of trips can't reach 600+ then there could be a significant difficuties. 

Another question is, used your new fixed and splitted file to run the code. From trace, I found that the generator still send tuples whose destination and arrival time is removed? Will they be taken accounted for evaluation?

Thank you very much!

Best regards,
D

**Comment by smirnp** [comment]
Hi,
1. New ports have been added to split and correctly evaluate such a valuable trajectories: https://gyazo.com/7642292dabe0a81ab024da9c4ad0ac03 https://gyazo.com/971b14adb30d1815bb4732febb5bc5c4
https://gyazo.com/01262d2f7ffb97548f18dae832e3a636
Otherwise they should be thrown away, because ships are parking in unknown places for weeks or months and then continue under the same DeparturePortName value.
2. No. The splitting/labelling algorithms are performing over the whole dataset, which is then splitted on training and test ones.
3. The overall trajectories of ships are remaining the same, but become splitted per newly added ports to be evalutated correctly. So is it seems to me, there is no need in a new data for training. 
4. Tuples without labels will be ignored in evaluation. But they still contribute to system's processing time, which is consideted in total rank.

**Comment by acip** [comment]
Just to (over)clarify, as this is very important for a couple of solution approaches:

> The overall trajectories of ships are remaining the same, but become splitted per newly added ports to be evalutated correctly. So is it seems to me, there is no need in a new data for training.

Does this mean that some ports may be included in evaluation (expected as prediction), but not included in the training set?

**Comment by nduy** [comment]
Thank you @smirnp. Expecting your response to @acip question...

**Comment by smirnp** [comment]
Hi all! 

Considering your comments during yesterday - the datasets will be labeled using the original 25-ports file. So, the extended file is decided is used only for splitting complex trips. 
Trips to unknown ports will be ignored. Trips from unknown ports to known ports will be labeled. 

> Does this mean that some ports may be included in evaluation (expected as prediction), but not included in the training set?

No. All 25 ports are covered in training dataset.

**Comment by rivetti** [comment]
Hi Pavel,

Do I understand correctly that the training and testing data is actually data containing trajectories involving 40 ports where 15 ports have been "removed"?

If it is so, I fail to understand the reason to introduce this bias that, in turn, generates trips to and from unknown ports, as well as other artifacts in the trajectories (such as vessel waiting in the middle of nowhere).

Thanks
Nicolo

**Comment by smirnp** [comment]
Hi Nicolo,

> Do I understand correctly that the training and testing data is actually data containing trajectories involving 40 ports where 15 ports have been "removed"?

Initially we decided to limit a potential prediction variability (space) with 25 ports and the the major part of the trips in dataset covers there 25 ports. But visual inspection showed, that also many trips are passing throught the another ports (and have parkings there), so it make sense to consider them for splitting. I haven't compared, which of the new ports are visited mostly, but the amount of visits is definetely less than for the ports listed in 25.  
Considering new ports for predictions we increase the general prediction space (complexity), but since we have a limited dataset the overall precision of the systems will degrade. Please describe your point, if you find this wrong.

**Comment by rivetti** [comment]
Hi Pavel,

I don't have now access to our system, so I will provide a made-up example.

For instance, if in the original data there are two trajectories: 
A: Barcellona - Castellon - Palma de Maiorca
B: Barcelona - Ibiza - Palma de Maiorca 
If we simply "hide" Castellon and Ibiza, we end up with two trajectories Barcellona - Palma de Maiorca, with totally different path, time to travel, etc.
Moreover, the vessels may park a while into Castellon and/or Ibiza, resulting in a vessel waiting in the midst of the sea.

If it occurs often, I wonder what affects more the precision: search space size or arbitrary noise added to the dataset.
If it does not occur often, then maybe an alternative solution is to not only remove the ports, but the whole "bad" trajectories.

Thanks
Nicolo

**Comment by bdnf** [comment]
That's a valid point made by Nicolo.

Moreover, there are other interesting trips, for example:

Livorno -> For Sur Mer -> Barcelona

Between "Livorno" and "For Sur Mer" the ship makes a stop in "Genova" (around 20 minutes during which REPORTED_DRAUGHT was increased by the value of 4.0). 
"Genova" - is the port that is present in the list of ports, however, this port is also 'hidden' from us, similarly to the examples above.

On the one hand, following the initial rules of the Challenge - it is possible to deal with such issues!
Throwing out such "bad" part of the data isn't a solution.
On the other - "hiding" of such ports might not resemble a real-world system in the end.

**Comment by smirnp** [comment]
Please find the [updated dataset](https://hobbitdata.informatik.uni-leipzig.de/DEBS_GC_2018/DEBS%202018_training_fixed_7.ZIP) with the extended ports list (44 items), as well as vizualizations of labeled/unlabeled trips. Currently 34 of ports are presented in the `arrival_port_calc` field in the training and test dataset.

**Comment by nduy** [comment]
Thank you @smirnp and others for the discussion. I share the concern with @rivetti and @bdnf that excluding unclear data is not necessarily a good choice, especially when we have limited number of trajectories. 

 Pavel, please let us know your final decision, i.e. whether this (v.7) is the final version of the dataset? Are you going to keep going on with the hidden/unknown port? I looking forward to heading your decision because our deadline is only a week from now...

**Comment by smirnp** [comment]
Hi, @nduy , 

Regarding data labelling algorithms- let's stop at this state (v.7) data until the submission deadline. 

For those, who submitted a working system into the platform we have prepared an additional piece of training data labelled with the same algorithm as the first training one (currently by the most reliable - v.7). The provider of the data allowed us to release the second dataset in such way only. To obtain the second training dataset (protected by another password), please email us with a link to any successful run of your system in the online platform.  The test data will be composed of two datasets. 

For those participants who are willing to continue to improve/tune their systems until the final evaluation (end of May) we might continue to improve data labelling algorithms and release updates with increased number of labelled trips (now there are some amount of incorrectly labelled/correct but not labelled trips). Registered systems will be executed automatically by some period (now 3 days) with the updates of the online leader board.

**Comment by nduy** [comment]
Got it, @smirnp. I have just found an issue with version v.7. There is a minor issue, that some tuples has REPORTED_DRAUGHT value "ITALY"""  instead of a number. It's line 399728 to 400775. There could be more similar issues, so please help us fix this. This is an example:

> 0xcf531a03076f130f93037caf7d5c55a3a17d6536,37,13.4,-5.342212,36.10098,112,104,04-05-15 11:59,GIBRALTAR,"ITALY""",0xcf531_04-05-15 11:xx - 07-05-15 8:34,07-05-15 7:46,GENOVA

**Comment by bdnf** [comment]
Hi @smirnp 
So, the system for submission deadline should be based on the previous state of the dataset or on this latest state (v.7)?

**Comment by smirnp** [comment]
Hi @bdnf ,
No, registration and submission deadlines cannot be extended anymore. But you may update images of registered system until the final run (end of May).

**Comment by smirnp** [comment]
@nduy , thanks! Fixed.

**Comment by clin99** [comment]
Hi @smirnp,  could you please confirm which port list will be used in test data set?  The extended one (40 ports) or the original one (25 ports)?

Thanks.

**Comment by smirnp** [comment]
Hi @clin99 , the extended one (40 ports)

**Comment by nduy** [comment]
But @smirnp, in the training + secondary datasets, only 33 destinations come as departure or arrival port.

**Comment by smirnp** [comment]
@nduy , yes that is correct that not all listed ports were covered by labels (at the current state in labeling algorithms)

**Comment by bdnf** [comment]
Thanks @smirnp  a lot for this enormous work with the dataset. Data is now much easier to work with!

Could it be possible that during final testing previously unseen ports would be used (those 7 not covered yet by labels)?
